### PR TITLE
Move piggy-back.py to diagnostics tree

### PR DESCRIPTION
**Pull request by nbren12**
Prepare piggy-back.py for further development. Currently there is no framework for testing this code. Once this refactor is merged, I can add some new plots comparing the condensation and precip skill.

Move this script into the `diagnostics` package and test it.

Added Public API:
- vcm.cdl_to_dataset. This is makes it easy to generate netCDF-like datasets. Common Data Language (CDL) is much more concise than the yaml format used by `synth`.
- CLI: `projects/microphysics/scripts/piggy-back.py` renamed to `prognostic_run_diagnostics piggy`

### CODE REVIEW

File: external/vcm/vcm/cdl.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
+"""Routines for working with the UCAR Common Data Language (CDL)"""
```
Comment (AnnaKwa) [inline external/vcm/vcm/cdl.py:1]: Can you add a link to the UCAR site with an example string?

**Review by AnnaKwa — APPROVED** [review]
LGTM!

File: conda-linux-64.lock [hunk 2]
```diff
@@ -65,26 +65,28 @@ https://conda.anaconda.org/conda-forge/linux-64/xorg-fixesproto-5.0-h7f98852_100
...
 https://conda.anaconda.org/conda-forge/linux-64/libcblas-3.9.0-13_linux64_openblas.tar.bz2#b17676dbd6688396c3a3076259fb7907
 https://conda.anaconda.org/conda-forge/linux-64/liblapack-3.9.0-13_linux64_openblas.tar.bz2#018b80e8f21d8560ae4961567e3e00c9
-https://conda.anaconda.org/conda-forge/linux-64/libnghttp2-1.46.0-h812cca2_0.tar.bz2#507fa47e9075f889af8e8b72925379be
+https://conda.anaconda.org/conda-forge/linux-64/libnghttp2-1.46.0-ha19adfc_0.tar.bz2#55e1d8f951e166199d7ab2ffdb75a497
 https://conda.anaconda.org/conda-forge/linux-64/libpng-1.6.37-h21135ba_2.tar.bz2#b6acf807307d033d4b7e758b4f44b036
-https://conda.anaconda.org/conda-forge/linux-64/libssh2-1.10.0-ha56f1ee_2.tar.bz2#6ab4eaa11ff01801cffca0a27489dc04
+https://conda.anaconda.org/conda-forge/linux-64/libssh2-1.10.0-ha35d2d1_2.tar.bz2#e0adb0915b3e6971a84e347f998ca837
 https://conda.anaconda.org/conda-forge/linux-64/libtiff-4.3.0-h6f004c6_2.tar.bz2#34fda41ca84e67232888c9a885903055
 https://conda.anaconda.org/conda-forge/linux-64/libxml2-2.9.12-h885dcf4_1.tar.bz2#d1355eaa48f465782f228275a0a69771
+https://conda.anaconda.org/conda-forge/linux-64/libzip-1.8.0-h1c5bbd1_1.tar.bz2#800eab3dd173ff4e48bc2e7cc5fa58e1
```
Comment (nbren12) [inline conda-linux-64.lock:76]: BSD-3

**Review by nbren12 — COMMENTED** [review]
license

**Pull request merged**